### PR TITLE
1198 [build] Add array utility to type system

### PR DESCRIPTION
**Pull request by Mearman**
closes https://github.com/breadboard-ai/breadboard/issues/1198

### CODE REVIEW

File: packages/build/src/test/type_test.ts [hunk 1]
```diff
@@ -185,3 +183,44 @@ test("unsafeType", () => {
...
+describe("array", () => {
+  test("no arguments", () => {
+    // @ts-expect-error no
+    assert.throws(() => array());
+  });
+
+  test("array of numbers", () => {
+    const arr1 = array("number");
+    // $ExpectType number[]
+    type t1 = ConvertBreadboardType<typeof arr1>;
```
Comment (aomarks) [inline packages/build/src/test/type_test.ts:196]: For these `defined but never used` errors you can add `/* eslint-disable @typescript-eslint/no-unused-vars */` blocks like in some of the above tests.

**Review by aomarks — APPROVED** [review]
Looks perfect! Just some minor lint errors caught by CI.

File: packages/build/src/internal/type-system/array.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  type ConvertBreadboardType,
+} from "./type.js";
+
+/**
+ * Creates an array type for the specified element type.
+ *
+ * @param elementType - The type of elements in the array.
+ * @returns An `AdvancedBreadboardType` representing an array of the specified element type.
+ */
+
```
Comment (aomarks) [inline packages/build/src/internal/type-system/array.ts:20]: Remove this newline

File: packages/build/src/test/type_test.ts (hunk 1 above)
Comment (Mearman) [reply packages/build/src/test/type_test.ts:196]: ah I even saw yours, and then completely forgot

**Pull request merged**